### DISCUSSION TITLE
How to get the handle to the root object from a group object?

### DISCUSSION THREAD

**Discussion by jgostick** [zarr-developers/zarr-python#1491] [Q&A]
Sorry for the noob question, but let's say I have created the following:

``` python
# Top level group
root = zarr.group()
g1 = root.create_group('papa')
g1['some data'] = np.ones(10)

# Now a sub-group
g2 = g1.create_group('son')
g2['more data'] = np.arange(20)

```

Now my question is: if I only have the handle to ``g2`` (like it gets passed into a function) can I find ``g1``?  I'm imagining a syntax like this:

``` python
def func(g2):
    g1 = g2.get_handle_to_root_group_which_I_belong()
    # some stuff with g1 and g2 data
    return g2
```

You may be wondering why I don't just pass `g1` into the function.  I am evaluating ZARR for use in our project [OpenPNM](https://openpnm.org), and it could actually work as a drop-in replacement if this is possible.

Thanks for your time and thanks for sharing this impressive library.

**Comment by rabernat** [comment]
This is a good question @jgostick. The API does not currently offer a way to do this afaik. But that would be a very reasonable feature request.
